src/main/java/org/elasql/util/FeatureCollector.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright 2016, 2018 vanilladb.org contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.elasql.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FeatureCollector {
+
+	private static final ThreadLocal<FeatureCollector> LOCAL_FEATURE_COLLECTOR = new ThreadLocal<FeatureCollector>() {
+		@Override
+		protected FeatureCollector initialValue() {
+			return new FeatureCollector();
+		}
+	};
+
+	/**
+	 * Get the feature collector local to this thread.
+	 * 
+	 * @return the local feature collector
+	 */
+	public static FeatureCollector getLocalFeatureCollector() {
+		return LOCAL_FEATURE_COLLECTOR.get();
+	}
+	
+	public static String[] keys = {"Start time", "Read PK", "Write PK"};

```

Comment:
I suggest to use `ArrayList<String>` here because it is much flexible and has the same performance with an array. In order to do that, you may write some code like:

```java
private static final List<String> keys;

static {
   keys = new ArrayList<String>();
   keys.add("Start Time");
   keys.add("Read PK");
   ...
}
```